plugins/embed-optimizer/class-embed-optimizer-tag-visitor.php

```diff
@@ -119,6 +158,80 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 			$this->added_lazy_script = true;
 		}
 
-		return true;
+		/*
+		 * At this point the tag is a figure.wp-block-embed, and we can return false because this does not need to be
+		 * measured and stored in URL Metrics. Only the child div.wp-block-embed__wrapper tag is measured and stored
+		 * so that this visitor can look up the height to set as a min-height on the figure.wp-block-embed. For more
+		 * information on what the return values mean for tag visitors, see <https://github.com/WordPress/performance/issues/1342>.
+		 */
+		return false;
+	}
+
+	/**
+	 * Gets the XPath for the embed wrapper.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_Tag_Visitor_Context $context Tag visitor context.
+	 * @return string XPath.
+	 */
+	private function get_embed_wrapper_xpath( OD_Tag_Visitor_Context $context ): string {
+		return $context->processor->get_xpath() . '/*[1][self::DIV]';
```

Comment:
What does this additional string at the end here do?

Reply:
Given an embed block such as the following:

```html
<figure class="wp-block-embed is-type-video is-provider-wordpress-tv wp-block-embed-wordpress-tv wp-embed-aspect-16-9 wp-has-aspect-ratio">
	<div class="wp-block-embed__wrapper">
		<iframe title="VideoPress Video Player" aria-label='VideoPress Video Player' width='750' height='422' src='https://video.wordpress.com/embed/vaWm9zO6?hd=1&amp;cover=1' frameborder='0' allowfullscreen allow='clipboard-write'></iframe>
		<script src='https://v0.wordpress.com/js/next/videopress-iframe.js?m=1674852142'></script>
	</div>
</figure>
```

If this were a direct descendant of the `BODY`, then the XPath for the `FIGURE` tag here would be:

```
/*[1][self::HTML]/*[2][self::BODY]/*[1][self::FIGURE]
```

What this method is doing is appending to this XPath the string to select the `div.wp-block-embed__wrapper` child element, which has the XPath of:

```
/*[1][self::HTML]/*[2][self::BODY]/*[1][self::FIGURE]/*[1][self::DIV]
```

I've clarified this in 455ef4fe.

I played around with drafting https://github.com/WordPress/performance/pull/1407 to add some helpers to construct XPaths which we can consider later instead of manually constructing XPath expressions.

Reply:
Makes sense, thanks for clarifying!